pkg/operator/ceph/disruption/clusterdisruption/reconcile.go

```diff
@@ -262,7 +262,17 @@ func (r *ReconcileClusterDisruption) deleteDrainCanaryPods(namespace string) err
 }
 
 func (r *ReconcileClusterDisruption) deleteLegacyPDBForOSD(namespace string) error {
-	err := r.client.DeleteAllOf(context.TODO(), &policyv1beta1.PodDisruptionBudget{}, client.InNamespace(namespace),
+	var podDisruptionBudget client.Object
+	usePDBV1Beta1, err := k8sutil.UsePDBV1Beta1Version(r.context.ClusterdContext.Clientset)
+	if err != nil {
+		return errors.Wrapf(err, "failed to fetch pdb version")
+	}
+	if usePDBV1Beta1 {
+		podDisruptionBudget = &policyv1beta1.PodDisruptionBudget{}
+	} else {
+		podDisruptionBudget = &policyv1.PodDisruptionBudget{}
```

Comment:
Why not just doing

```go
podDisruptionBudget := &policyv1.PodDisruptionBudget{}
usePDBV1Beta1, err := k8sutil.UsePDBV1Beta1Version(r.context.ClusterdContext.Clientset)
if err != nil {
    return errors.Wrapf(err, "failed to fetch pdb version")
}
if usePDBV1Beta1 {
    podDisruptionBudget = &policyv1beta1.PodDisruptionBudget{}
}
```

Instead? So once we don't support beta anymore you can remove all the code after `podDisruptionBudget := &policyv1.PodDisruptionBudget{}`.

Reply:
Okay let me try, 
But I guess if we use the k8s version lesser than 1.21.0, the line `podDisruptionBudget := &policyv1.PodDisruptionBudget{}` will through an error as the v1 version is not supported. 

And if we create a  `podDisruptionBudget = &policyv1beta1.PodDisruptionBudget{}` first then ig it will work.
But will again cause re-structuring afterward.

Reply:
Right, different type, my bad.